File: Angular/fileshare/src/app/classes/comment.ts
import {File} from "./file";
import {User} from "./user";

export class Comment {

  private _id: number;
  private _file: File;
  private _text: string;
  private _user: User;
  private _date: Date;

  public constructor(id: number, file: File, text: string, user: User, date: Date) {
    this._id = id;
    this._file = file;
    this._text = text;
    this._user = user;
    this._date = date;
  }

  public get id(): number {
    return this._id;
  }

  public get file(): File {
    return this._file;
  }

  public get text(): string {
    return this._text;
  }

  public get user(): User {
    return this._user;
  }

  public get date(): Date {
    return this._date;
  }
}

File: fileshare/src/main/java/alkfejl/bead/fileshare/model/Comment.java
package alkfejl.bead.fileshare.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import org.springframework.beans.factory.annotation.Autowired;

import javax.persistence.*;
import java.util.Date;

@Entity
@Table(name = "COMMENT")
@Data
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class Comment extends BaseEntity {

    /*@JoinColumn
    @ManyToOne(targetEntity = User.class)
    private User reporter;
*/
    @JoinColumn
    @ManyToOne(targetEntity = File.class)
    private File commentedFile;

    @Column(nullable = false)
    private String text;

    @JoinColumn(nullable = false)
    @ManyToOne(targetEntity = User.class)
    private User user;

    @Column
    private Date date = new Date();

    @Override
    public String toString() {
        return this.date + " - " + this.user + ": " + this.text;
    }
}